pkgs/build-support/fetchmavenartifact/default.nix

```diff
@@ -0,0 +1,61 @@
+# Adaptation of the MIT-licensed work on `sbt2nix` done by Charles O'Farrell
+
+{ fetchurl, stdenv }:
+let
+  defaultRepos = [
+    http://central.maven.org/maven2
+    http://oss.sonatype.org/content/repositories/releases
+    http://oss.sonatype.org/content/repositories/public
+    http://repo.typesafe.com/typesafe/releases
+  ];
+in
+
+args@
+{ # Example: "org.apache.httpcomponents"
+  groupId
+, # Example: "httpclient"
+  artifactId
+, # Example: "4.3.6"
+  version
+, # Maven repository from where to fetch the artifact (example:
+  # http://oss.sonatype.org/content/repositories/public)
+  repo ? null
+, # List of maven repositories from where to fetch the artifact.
+  # Has priority over `repo`.
```

Comment:
This is exactly the same behavior fetchurl already has. Like you, I'm not
sure if it is the best approach, but it is the most conservative given
fetchurl's current behavior, I think.

We could just drop the repo parameter and keep the repos parameter. I'd be
happy with that.
On Jul 10, 2016 2:12 AM, "Alexey Shmalko" <notifications@github.com> wrote:

> In pkgs/build-support/fetchmavenartifact/default.nix
> <https://github.com/NixOS/nixpkgs/pull/16825#discussion_r70176091>:
>
> > +    http://repo.typesafe.com/typesafe/releases
> > +  ];
> > +in
> > +
> > +args@
> > +{ # Example: "org.apache.httpcomponents"
> > +  groupId
> > +, # Example: "httpclient"
> > +  artifactId
> > +, # Example: "4.3.6"
> > +  version
> > +, # Maven repository from where to fetch the artifact (example:
> > +  # http://oss.sonatype.org/content/repositories/public)
> > +  repo ? null
> > +, # List of maven repositories from where to fetch the artifact.
> > +  # Has priority over `repo`.
>
> That's unfortunate, as now you must override both repo and repos. And the
> following expression doesn't work and may cause a happy debugging:
>
> fetchMavenArtifact {
>   groupId = "...";
>   artifactId = "...";
>   version = "...";
>   repo = "...";
> }
>
> I propose to either make a repo a higher priority or drop this parameter.
>
> fetchMavenArtifact {
>   groupId = "...";
>   artifactId = "...";
>   version = "...";
>   repos = [ "..." ];
> }
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/NixOS/nixpkgs/pull/16825/files/8ca6205d3d301215c34029629ebafe8fd0b42136#r70176091>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AAAM5mi7O8jbpe5orDOVbtnGjzULKdAwks5qUH9PgaJpZM4JIuAP>
> .
>